packages/gatsby/src/utils/get-ssl-cert.js

```diff
@@ -1,9 +1,29 @@
 const getDevelopmentCertificate = require(`devcert-san`).default
 const report = require(`gatsby-cli/lib/reporter`)
+const fs = require(`fs`)
 
-module.exports = async name => {
-  report.info(`setting up SSL certificate (may require sudo)\n`)
+module.exports = async ({ name, certFile, keyFile, directory }) => {
+  // check that cert file and key file are both true or both false, if they are both
+  // false, it defaults to the automatic ssl
+  if (certFile ? !keyFile : keyFile) {
+    report.panic(
+      `for custom ssl --https, --cert-file, and --key-file must be used together`
+    )
+  }
+
+  if (certFile && keyFile) {
+    const keyPath = `${directory}/${keyFile}`
+    const certPath = `${directory}/${certFile}`
```

Comment:
maybe use `path.join` instead of manual string concat for paths? 
this also forces relative path right now so probably should be noted in docs?